### PR TITLE
adding views for creating, updating, deleting comments

### PR DESCRIPTION
**Pull request by marinaRomanchuk**

### CODE REVIEW

File: comments/serializers.py [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
-    users = UserPublicSerializer(many=True, read_only=True)
-
     class Meta:
         model = Comment
-        fields = ["text", "post_id", "users", "timestamp"]
+        fields = ["text", "post", "timestamp"]
+
+    def create(self, data: Dict) -> Comment:
+        request = self.context.get("request")
+        comment = Comment.objects.create(
```
Comment (revimi) [inline comments/serializers.py:15]: `.objects.create` calls `.save()`. You don't have to call it explicitly

File: comments/urls.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        "get": "retrieve",
+        "put": "update",
+        "patch": "partial_update",
+        "delete": "destroy",
+    }
+)
+
+urlpatterns = [
+    url(r"^comments/add/$", comment_add, name="add-comment"),
```
Comment (revimi) [inline comments/urls.py]: We don't need `add` postfix. Use `POST` to `r"^comments/$"`

File: comments/views.py [hunk 3]
```diff
@@ -1,3 +1,30 @@
...
+    def list(self, request) -> Response:
+        queryset = Comment.objects.all()
+
+        try:
+            post_id: Union[int, None] = int(self.request.GET.get("post_id"))
+        except (ValueError, TypeError):
+            post_id = None
+
+        if post_id:
+            queryset = queryset.filter(post_id=post_id)
```
Comment (revimi) [inline comments/views.py:25]: :+1:

File: comments/views.py [hunk 4]
```diff
@@ -1,3 +1,30 @@
...
+        else:
+            queryset = queryset.filter(user=request.user)
```
Comment (revimi) [inline comments/views.py]: What is the use case for this?

**Review by revimi — COMMENTED** [review]
Looks good! few minor comments

File: comments/serializers.py (hunk 1 above)
Comment (marinaRomanchuk) [reply comments/serializers.py:15]: Removed

File: comments/urls.py (hunk 2 above)
Comment (marinaRomanchuk) [reply comments/urls.py]: `r"^comments/$"` can be used to create a comment now

File: comments/views.py (hunk 4 above)
Comment (marinaRomanchuk) [reply comments/views.py]: Removed, I had doubts about this case

**Review by revimi — APPROVED** [review]

**Pull request merged**